components/filter-field/src/filter-field.ts

```diff
@@ -373,8 +383,16 @@ export class DtFilterField<T> implements AfterViewInit, OnDestroy, OnChanges {
   ) {
     this.errorStateMatcher = defaultErrorStateMatcher;
     this._stateChanges
-      .pipe(switchMap(() => this._zone.onMicrotaskEmpty.pipe(take(1))))
+      .pipe(
+        switchMap(() => this._zone.onMicrotaskEmpty.pipe(take(1))),
+        takeUntil(this._destroy$),
```

Comment:
NIT: no takeUntil(destroy) required here because stateChanges completes when the component is destroyed